src/consensus/replay_tower.zig

```diff
@@ -1579,6 +1613,622 @@ fn optimisticallyBypassVoteStakeThresholdCheck(
+        if (key.equals(vote_account_pubkey)) {
+            my_latest_landed_vote = if (vote_state.nthRecentLockout(0)) |l| l.slot else null;
+            logger.debug().logf("vote state {any}", .{vote_state});
+            const observed_slot = if (vote_state.nthRecentLockout(0)) |l| l.slot else 0;
+
+            logger.debug().logf("observed slot {any}", .{observed_slot});
+        }
+        const start_root = vote_state.root_slot;
+
+        // Add the last vote to update the `heaviest_subtree_fork_choice`
+        if (vote_state.lastVotedSlot()) |last_landed_voted_slot| {
+            if (progress_map.getHash(last_landed_voted_slot)) |frozen_hash| {
+                _ = try latest_validator_votes_for_frozen_banks.checkAddVote(
+                    allocator,
+                    key,
+                    last_landed_voted_slot,
+                    frozen_hash,
+                    .replay,
+                );
+            }
+        }
+
+        // Simulate next vote and extract vote slots using the provided bank slot.
+        try vote_state.processNextVoteSlot(bank_slot);
+
+        for (vote_state.votes.constSlice()) |vote| {
+            try vote_slots.put(vote.slot);
+        }
+
+        if (start_root != vote_state.root_slot) {
+            if (start_root) |root| {
+                const vote = Lockout{ .slot = root, .confirmation_count = MAX_LOCKOUT_HISTORY };
+                logger.trace().logf("ROOT: {}", .{vote.slot});
+                try vote_slots.put(vote.slot);
+            }
+        }
+        if (vote_state.root_slot) |root| {
+            const vote = Lockout{ .slot = root, .confirmation_count = MAX_LOCKOUT_HISTORY };
+            try vote_slots.put(vote.slot);
+        }
+
+        // The last vote in the vote stack is a simulated vote on bank_slot, which
+        // we added to the vote stack earlier in this function by calling processVote().
+        // We don't want to update the ancestors stakes of this vote b/c it does not
+        // represent an actual vote by the validator.
+
+        // Note: It should not be possible for any vote state in this bank to have
+        // a vote for a slot >= bank_slot, so we are guaranteed that the last vote in
+        // this vote stack is the simulated vote, so this fetch should be sufficient
+        // to find the last unsimulated vote.
+        std.debug.assert(vote_state.nthRecentLockout(0).?.slot == bank_slot);
+
+        if (vote_state.nthRecentLockout(1)) |vote| {
+            // Update all the parents of this last vote with the stake of this vote account
+            try updateAncestorVotedStakes(
+                allocator,
+                &voted_stakes,
+                vote.slot,
+                voted_stake,
+                ancestors,
+            );
+        }
+        total_stake += voted_stake;
+    }
+
+    try populateAncestorVotedStakes(
+        allocator,
+        &voted_stakes,
+        vote_slots.items(),
+        ancestors,
+    );
```

Comment:
```suggestion
    try populateAncestorVotedStakes(allocator, &voted_stakes, vote_slots.items(), ancestors);
```

is this an accident, or do you prefer to wrap some code when it's not necessary?

Reply:
It is this one that was actually an accident https://github.com/Syndica/sig/pull/798#discussion_r2226298255. I generally format into newlines when the input params is 2 or more.

Reply:
Interesting ok. Normally I prefer to keep these on one line, but I guess it's a matter of personal preference. I'm fine with this as is